pkg/embed/embed.go

```diff
@@ -6,20 +6,43 @@ import (
 	"os"
 
 	"github.com/code-ready/crc/pkg/crc/logging"
-
-	"github.com/YourFin/binappend"
 )
 
-func openEmbeddedFile(executablePath, embedName string) (*binappend.Reader, error) {
-	extractor, err := binappend.MakeExtractor(executablePath)
-	if err != nil {
-		return nil, fmt.Errorf("Could not data embedded in %s: %v", executablePath, err)
-	}
-	reader, err := extractor.GetReader(embedName)
-	if err != nil {
-		return nil, fmt.Errorf("Could not open embedded '%s' in %s: %v", embedName, executablePath, err)
+//go:generate go run generator.go
+
+type embedFiles struct {
+	storage map[string][]byte
+}
+
+// Create new box for embed files
+func newEmbedBox() *embedFiles {
+	return &embedFiles{storage: make(map[string][]byte)}
+}
+
+// Add a file to box
+func (e *embedFiles) Add(file string, content []byte) {
+	e.storage[file] = content
+}
+
+// Get file's content
+func (e *embedFiles) Get(file string) []byte {
+	if f, ok := e.storage[file]; ok {
+		return f
 	}
-	return reader, nil
+	return nil
+}
+
+// Embed box expose
+var box = newEmbedBox()
```

Comment:
Not really sold on the `box` naming, is this naming commonly used in other projects?

Reply:
@cfergeau this is something which packr uses https://github.com/gobuffalo/packr#in-code so I tried to use the same.